core/services/assets/AssetRegisterInterface.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace EventEspresso\core\services\assets;
+
+defined('EVENT_ESPRESSO_VERSION') || exit;
+
+
+
+/**
+ * Interface AssetRegisterInterface
+ * Defines any class capable of handling asset registration via EventEspresso\core\services\assets\Registry
+ *
+ * @package EventEspresso\core\services\assets
+ * @author  Brent Christensen
+ * @since   $VID:$
+ */
+interface AssetRegisterInterface
+{
+    /**
+     * @return  void
+     */
+    public function registerManifestFile();
+
+    /**
+     * @return  void
+     */
+    public function registerScripts();
+
+    /**
+     * @return void
+     */
+    public function registerStyles();
+
+    /**
+     * @return string
+     */
+    public function scriptHandle();
+
+    /**
+     * @return string
+     */
+    public function styleHandle();
```

Comment:
So based on my previous comment, I wonder if we should either have specific methods for `editorscript` and `editorstyle` to differentiate from just `script` and `style` or just use a flag in these methods to indicate context for retrieval?

Reply:
I think I'll go with the former, since it's just more explicit imho